src/@apollo/client/react/hooks/ApolloClient__React_UseMutation.re

```diff
@@ -0,0 +1,368 @@
+    ~update=?,
+    ~variables=?,
+    (module OperationNoRequiredVars),
+  ) => {
+    let jsMutationTuple =
+      Js_.useMutation(.
+        ~mutation=
+          OperationNoRequiredVars.query->Utils.castStringAsDocumentNode,
+        ~options=
+          MutationHookOptions.toJs(
+            {
+              mutation: None,
+              awaitRefetchQueries,
+              context,
+              client,
+              errorPolicy,
+              fetchPolicy,
+              ignoreResults,
+              notifyOnNetworkStatusChange,
+              onError,
+              onCompleted,
+              optimisticResponse,
+              refetchQueries,
+              update,
+              variables:
+                Some(
+                  variables->Belt.Option.getWithDefault(
+                    OperationNoRequiredVars.makeDefaultVariables(),
+                  ),
+                ),
+            },
+            ~parse=OperationNoRequiredVars.parse,
+            ~serialize=OperationNoRequiredVars.serialize,
+          ),
+      );
+
+    Utils.useGuaranteedMemo1(
+      () => {
+        jsMutationTuple->MutationTuple__optionalVariables.fromJs(
+          ~defaultVariables=OperationNoRequiredVars.makeDefaultVariables(),
+          ~parse=OperationNoRequiredVars.parse,
+          ~serialize=OperationNoRequiredVars.serialize,
+        )
+      },
+      jsMutationTuple,
+    );
+  };
+
+module Extend = (M: Types.Operation) => {
+  let use =
+      (
+        ~awaitRefetchQueries=?,
+        ~context=?,
+        ~client=?,
+        ~errorPolicy=?,
+        ~fetchPolicy=?,
+        ~ignoreResults=?,
+        ~notifyOnNetworkStatusChange=?,
+        ~onError=?,
+        ~onCompleted=?,
+        ~optimisticResponse=?,
+        ~refetchQueries=?,
+        ~update=?,
+        (),
+      ) => {
+    useMutation(
+      ~awaitRefetchQueries?,
+      ~context?,
+      ~client?,
+      ~errorPolicy?,
+      ~fetchPolicy?,
+      ~ignoreResults?,
+      ~notifyOnNetworkStatusChange?,
+      ~onError?,
+      ~onCompleted?,
+      ~optimisticResponse?,
+      ~refetchQueries?,
+      ~update?,
+      (module M),
+    );
+  };
+
+  let useWithVariables =
+      (
+        ~awaitRefetchQueries=?,
+        ~context=?,
+        ~client=?,
+        ~errorPolicy=?,
+        ~fetchPolicy=?,
+        ~ignoreResults=?,
+        ~notifyOnNetworkStatusChange=?,
+        ~onError=?,
+        ~onCompleted=?,
+        ~optimisticResponse=?,
+        ~refetchQueries=?,
+        ~update=?,
+        variables,
+      ) => {
+    useMutationWithVariables(
+      ~awaitRefetchQueries?,
+      ~context?,
+      ~client?,
+      ~errorPolicy?,
+      ~fetchPolicy?,
+      ~ignoreResults?,
+      ~notifyOnNetworkStatusChange?,
+      ~onError?,
+      ~onCompleted?,
+      ~optimisticResponse?,
+      ~refetchQueries?,
+      ~update?,
+      ~variables,
+      (module M),
+    );
+  };
+};
+
+module ExtendNoRequiredVariables = (M: Types.OperationNoRequiredVars) => {
```

Comment:
@jfrolich, I did not follow the pattern established [here](https://github.com/reasonml-community/reason-apollo-hooks/pull/117/files#diff-5e4285d18908d1a3fbdf96584da5bf74R457) since it didn't seem to matter if you applied variables to the hook or returned mutation function, or both, but I might be missing something (I never use `useMutation`).

Reply:
It's a bit subtle, but if you have no required variables, the variables in `useWithVariables` are optional, if you don't pass in the mutation is called without variables, and the callback also doesn't have variables as an argument.

Example
```reason
let (setAsFavorite, _) = FavoriteMutation.useWithVariables();
``` 
now you can call `setAsFavorite()` (with no arguments).

Otherwise, for the same result without `ExtendNoRequiredVariables`:

```reason
let (setAsFavorite, _) = FavoriteMutation.useWithVariables(~variables=FavoriteMutation.makeDefaultVariables());
```

or 

```reason
let (setAsFavorite, _) = FavoriteMutation.use();
```

and then `setAsFavorite(~variables=FavoriteMutation.makeDefaultVariables(), ())`

Does that explain it?

Reply:
Totally. Thanks!